src/core/SampleBuffer.cpp

```diff
@@ -131,6 +131,81 @@ SampleBuffer::SampleBuffer( const f_cnt_t _frames )
 
 
 
+SampleBuffer::SampleBuffer(const SampleBuffer& orig)
+{
+	orig.m_varLock.lockForRead();
+
+	m_audioFile = orig.m_audioFile;
+	m_origData = MM_ALLOC(sampleFrame, orig.m_origFrames);
+	m_origFrames = orig.m_origFrames;
+	m_data = MM_ALLOC(sampleFrame, orig.m_frames);
+	m_frames = orig.m_frames;
+	m_startFrame = orig.m_startFrame;
+	m_endFrame = orig.m_endFrame;
+	m_loopStartFrame = orig.m_loopStartFrame;
+	m_loopEndFrame = orig.m_loopEndFrame;
+	m_amplification = orig.m_amplification;
+	m_reversed = orig.m_reversed;
+	m_frequency = orig.m_frequency;
+	m_sampleRate = orig.m_sampleRate;
+
+	//Deep copy m_origData and m_data from original
+	memcpy(m_origData, orig.m_origData, m_origFrames * BYTES_PER_FRAME);
+	memcpy(m_data, orig.m_data, m_frames * BYTES_PER_FRAME);
```

Comment:
If `orig.m_origData` or `orig.m_data` can be null, such as for a default-constructed `SampleBuffer`. Calling `memcpy` with a null pointer is undefined behaviour, even if the size is zero.

Reply:
In addition to the comment above, MOST implementations of `memcpy` does nothing if the length is 0, but it's not guaranteed that ALL implementations do nothing.

Reply:
Hopefully fixed in latest commit